src/main.rs

```diff
@@ -37,28 +37,47 @@ use alacritty::tty::{self, process_should_exit};
 use alacritty::util::fmt::Red;
 
 fn main() {
+    // Load command line options
+    let options = cli::Options::load();
 
     // Load configuration
-    let config = Config::load().unwrap_or_else(|err| {
-        match err {
-            // Use default config when not found
-            config::Error::NotFound => {
-                match Config::write_defaults() {
-                    Ok(path) => err_println!("Config file not found; write defaults config to {:?}", path),
-                    Err(err) => err_println!("Write defaults config failure: {}", err)
+    // If a configuration file is given as a command line argument we don't generate a default file.
+    // If an invalid configuration file is given, i.e. /dev/null, we load the compiled in defaults.
+    let config = match options.config {
```

Comment:
To prevent rightward-growth, and just for simplicity, I would try writing this out using combinators as

```rust
let config = options.config.as_ref()
    .and_then(Config::load_from)
    .or_else(|| /* normal load */);
```

That's not quite the full story, but trying to finish a review before plane takes off 🙃